sdks/go/pkg/beam/runners/dataflow/dataflow.go

```diff
@@ -62,6 +62,7 @@ var (
 	minCPUPlatform       = flag.String("min_cpu_platform", "", "GCE minimum cpu platform (optional)")
 	workerJar            = flag.String("dataflow_worker_jar", "", "Dataflow worker jar (optional)")
 
+	async          = flag.Bool("async", false, "Asynchronous execution. Submit the job and return immediately.")
```

Comment:
Consider prefixing this as dataflow_async or similar to avoid collisions since async is very general. (dry_run also runs into this problem, but hasn't been a priority to change).

Reply:
How about `execute_async`?

Reply:
I think I prefer `execute_async`, as it could be useful for runners other than dataflow, but happy to go with whichever the two of you think is preferable. Is there a desire to keep the flags consistent across the different SDKs? I couldn't find something comparable to this one to emulate, but the Python SDK has `dry_run` too, so if we rename that we'd be working against any such consistency.

Python has `wait_until_finish_duration`, which sounds similar to what I'm trying to do, but appears to cancel the pipeline if it hasn't executed within that window.

Reply:
I think `execute_async` convey the intent of the flag.
Python does not block the submission process unless we call `wait_until_finish_duration` in the code so the contract is other way around where we pass `wait_until_finish_duration` to make it synchronous while by default, the execution is async.
However, @lostluck its your call :)

Reply:
execute_async works for me!